lang/c++/impl/Compiler.cc

```diff
@@ -360,25 +396,37 @@ static NodePtr makeFixedNode(const Entity& e,
         throw Exception(boost::format("Size for fixed is not positive: ") %
             e.toString());
     }
-    return NodePtr(new NodeFixed(asSingleAttribute(name),
-        asSingleAttribute(v)));
+    NodePtr node =
+        NodePtr(new NodeFixed(asSingleAttribute(name), asSingleAttribute(v)));
+    if (containsField(m, "doc")) {
+        node->setDoc(getDocField(e, m));
+    }
+    return node;
 }
 
 static NodePtr makeArrayNode(const Entity& e, const Object& m,
     SymbolTable& st, const string& ns)
 {
     Object::const_iterator it = findField(e, m, "items");
-    return NodePtr(new NodeArray(asSingleAttribute(
-        makeNode(it->second, st, ns))));
+    NodePtr node = NodePtr(new NodeArray(
```

Comment:
According to [Avro spec](https://avro.apache.org/docs/1.8.1/spec.html) only records, fields and enums can have `doc`. That looks arbitrary. But until that is changed, we should not accept `doc` in other places. :). Interestingly, the Java code accepts doc in these places, but also in `fixed`. Maybe we can just do what the Java implementation does.

Reply:
This code does follow the Java implementation. Have I missed something?

Reply:
The only difference between Java and C++ implementation is that Java does not recognize `doc` for arrays and maps while this patch does.

Reply:
I'm happy to add support to Java in a follow up PR. Would it be fine to go ahead with this change?